src/StatusQ/Controls/StatusBaseInput.qml

```diff
@@ -45,11 +47,13 @@ Item {
     property bool valid: true
     property bool pristine: true
     property bool dirty: false
+    property bool leftIcon: true
```

Comment:
As mentioned offline, it's not clear to me why this is needed having an `icon.name` should indicate that there's an icon, no?

Reply:
I see, so this is to tell the component to not render the `icon` on the left side when a right side icon is needed.

There's indeed various other components that can potentially be rendered on the right side (e.g. there's a "Paste" button in some input components that need to be rendered in the same place).

I think ideally, we'd introduce a similar API to the one we have in `StatusListItem`, which is the `components` property (or just `component` in this case), which lets you render any kind of component in a particular slot:

```qml
StatusInput {
  ...
  component: StatusFlatRoundButton { ... } // Or StatusIcon or whatever else is needed
}
```
Having an API like that we:

1. Don't need to introduce a `leftIcon` property as this can be done based on heuristics
2. In fact, it's now possible to have both, the left icon **and** any component on the right hand side
3. Allow for other components than just icons (which we need anyways)

Do you feel like updating your PR to reflect such an API?
If it's too much of a hassle, we can do it afterwards as well, but I personally prefer to not introduce APIs that are known to be removed in the next step anyways.

Reply:
Created https://github.com/status-im/StatusQ/issues/380 accordingly

Reply:
> no, it's to define whether the icon should go in the left or right side. Should I rename it to isLeftIcon or something like this? Would that be better?

Figured it out eventually :D : https://github.com/status-im/StatusQ/pull/371#discussion_r702643686

You can rename it if you want, but as mentioned in the comment above, we'll probably remove it very soon again anyways. I'm working on stuff that requires a comopnent on the right hand side as well, so might as well create a proper API for that.

Reply:
Well, not sure if I should work on this on this occasion. I have already delayed my other task as you suggested to make use of StatusInput and the component was not supporting various requirements of it so I had to work on that as well. This is basically because when adding a new account we have to show a check mark when a seed phrase or private is entered correctly, so since it's about the same icon but different positions I thought to re-use the existing one instead of duplicating code. I am not aware of any other use cases of the StatusInput.

Reply:
Okay, that's fair. Will do it in the next iteration then.

>  I am not aware of any other use cases of the StatusInput.

See https://github.com/status-im/StatusQ/issues/380 for more information